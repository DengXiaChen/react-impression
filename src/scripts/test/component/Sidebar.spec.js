import React from 'react';
import { shallow, mount } from 'enzyme';
import { Sidebar } from '../../components/impression';

const setup = () => {
    const temp = (
        <Sidebar>
            <Sidebar.Header>
                <div>Sidebar header</div>
            </Sidebar.Header>
            <Sidebar.Body>
                <div>Sidebar body</div>
            </Sidebar.Body>
            <Sidebar.Footer className="test">
                <div>Sidebar footer</div>
            </Sidebar.Footer>
        </Sidebar>
    );
    const wrapper = mount(temp);
    const wrapper1 = shallow(temp);

    return {
        wrapper,
        wrapper1,
    };
};

describe('Sidebar', () => {
    const { wrapper, wrapper1 } = setup();

    it('render', () => {
        expect(wrapper.find('.sidebar').exists()).toBe(true);
        expect(wrapper.find('.sidebar-header').text()).toBe('Sidebar header');
        expect(wrapper.find('.sidebar-body').text()).toBe('Sidebar body');
        expect(wrapper.find('.sidebar-footer').text()).toBe('Sidebar footer');
    });

    it('ui', () => {
        expect(wrapper1).toMatchSnapshot();
    });
});
