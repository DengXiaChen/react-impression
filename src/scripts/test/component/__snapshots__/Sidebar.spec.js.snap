// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="sidebar"
>
    <SidebarHeader>
        <div>
            Sidebar header
        </div>
    </SidebarHeader>
    <SidebarBody>
        <div>
            Sidebar body
        </div>
    </SidebarBody>
    <SidebarFooter
        className="test"
    >
        <div>
            Sidebar footer
        </div>
    </SidebarFooter>
</div>,
  "nodes": Array [
    <div
      className="sidebar"
>
      <SidebarHeader>
            <div>
                  Sidebar header
            </div>
      </SidebarHeader>
      <SidebarBody>
            <div>
                  Sidebar body
            </div>
      </SidebarBody>
      <SidebarFooter
            className="test"
      >
            <div>
                  Sidebar footer
            </div>
      </SidebarFooter>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Sidebar>
        <SidebarHeader>
                <div>
                        Sidebar header
                </div>
        </SidebarHeader>
        <SidebarBody>
                <div>
                        Sidebar body
                </div>
        </SidebarBody>
        <SidebarFooter
                className="test"
        >
                <div>
                        Sidebar footer
                </div>
        </SidebarFooter>
</Sidebar>,
      "_debugID": 13,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            <SidebarHeader>
              <div>
                            Sidebar header
              </div>
</SidebarHeader>,
            <SidebarBody>
              <div>
                            Sidebar body
              </div>
</SidebarBody>,
            <SidebarFooter
              className="test"
>
              <div>
                            Sidebar footer
              </div>
</SidebarFooter>,
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="sidebar"
>
          <SidebarHeader>
                    <div>
                              Sidebar header
                    </div>
          </SidebarHeader>
          <SidebarBody>
                    <div>
                              Sidebar body
                    </div>
          </SidebarBody>
          <SidebarFooter
                    className="test"
          >
                    <div>
                              Sidebar footer
                    </div>
          </SidebarFooter>
</div>,
        "_debugID": 14,
        "_renderedOutput": <div
          className="sidebar"
>
          <SidebarHeader>
                    <div>
                              Sidebar header
                    </div>
          </SidebarHeader>
          <SidebarBody>
                    <div>
                              Sidebar body
                    </div>
          </SidebarBody>
          <SidebarFooter
                    className="test"
          >
                    <div>
                              Sidebar footer
                    </div>
          </SidebarFooter>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Sidebar>
    <SidebarHeader>
        <div>
            Sidebar header
        </div>
    </SidebarHeader>
    <SidebarBody>
        <div>
            Sidebar body
        </div>
    </SidebarBody>
    <SidebarFooter
        className="test"
    >
        <div>
            Sidebar footer
        </div>
    </SidebarFooter>
</Sidebar>,
}
`;
